images/windows/scripts/build/Install-PostgreSQL.ps1

```diff
@@ -58,13 +58,24 @@ if ($null -ne ($toolsetVersion | Select-String -Pattern '\d+\.\d+\.\d+')) {
     } while (!$response)
 }
 
+if ((Get-ToolsetContent).postgresql.installVcRedist) {
+    # Postgres 14 requires the vs 17 redistributable
+    $vs17RedistUrl = "https://aka.ms/vs/17/release/vc_redist.x64.exe"
+    Install-Binary `
+        -Url $vs17RedistUrl `
+        -InstallArgs @("/install", "/quiet", "/norestart") `
+        -ExpectedSignature (Get-ToolsetContent).postgresql.vcRedistSignature
+}
+
 # Return the previous value of ErrorAction and invoke Install-Binary function
 $ErrorActionPreference = $errorActionOldValue
 $installerArgs = @("--install_runtimes 0", "--superpassword root", "--enable_acledit 1", "--unattendedmodeui none", "--mode unattended")
+
 Install-Binary `
     -Url $installerUrl `
     -InstallArgs $installerArgs `
-    -ExpectedSignature (Get-ToolsetContent).postgresql.signature
+    -ExpectedSignature (Get-ToolsetContent).postgresql.signature `
+    -InstallerLogPath "$env:TEMP\**\install-postgresql.log"
```

Comment:
Using this I was able to find the issue, the installer errors with:

    azure-arm.image: The database cluster will be initialized with locale "English_United States.1252".
    azure-arm.image: The default text search configuration will be set to "english".
    azure-arm.image:
    azure-arm.image: Data page checksums are disabled.
    azure-arm.image: initdb: error: could not open file "C:\Users\packer\AppData\Local\Temp/postgresql_installer_b876155f3b\rad3B194.tmp" for reading: Permission denied
    azure-arm.image:
    azure-arm.image: Called Die(Failed to initialise the database cluster with initdb)...
    azure-arm.image: Failed to initialise the database cluster with initdb
    azure-arm.image:
    azure-arm.image: Script stderr:
    azure-arm.image:  Program ended with an error exit code
    azure-arm.image:
    azure-arm.image: Error running C:\Windows\System32\cscript //NoLogo "C:\Program Files\PostgreSQL\14/installer/server/initcluster.vbs" "NT AUTHORITY\NetworkService" "postgres" "****" "C:\Users\packer\AppData\Local\Temp/postgresql_installer_b876155f3b" "C:\Program Files\PostgreSQL\14" "C:\Program Files\PostgreSQL\14\data" 5432 "DEFAULT" 1: Program ended with an error exit code

https://github.com/actions/runner-images-ci/actions/runs/12159209946/job/33908835054#step:8:51246

Reply:
Looks like others are hitting this issue 

- https://github.com/EnterpriseDB/edb-installers/issues/196#issuecomment-2489021239

Reply:
EBD have a test installer which may resolve, looks like it's being actively worked on at least :crossed_fingers: